Automattic/woocommerce-payments
# Add/Update required components for progressive onboarding

Fixes #4744

#### Changes proposed in this Pull Request
Implement the following components following PO design.
- **LoadBar** – An infinite load bar.
- **PhoneNumberControl** – Implemented similar to Stripe KYC one, in absence of our own design. It could be replaced later by paJDYF-85M-p2.
  - Country codes coming from `intl-tel-input`.
  - Use native `select` and `input` elements.
  - Selecting a country or clicking on the country code switch focus to the input.
  - Defaults to `wcpaySetting.connect.country` or `US`
  - Returns a phone number in E164 format.
- **GroupedSelectControl**
  - Group provided options by `option.group` which should match `group.key`.
  - Allow to toggle groups visibility, showing the first one by default.
  - Optionally searchable, filtering by option name, and also by the optional property `context`. Restoring visibility state after clearing search.
  - Allow selection by both mouse and keyboard.

Update `CustomSelectControl` to match final design and implement `useUniqueId` hook using `lodash` under the hood.

#### Testing instructions

They should match our design and behave according to our needs.

>**Note** Those are just the components' implementation without using them directly. To check their functionality we need to add somewhere a code like the example below.

<details>
<summary>Example code</summary>

```javascript
import LoadBar from 'components/load-bar';
import PhoneNumberControl from 'components/phone-number-control';
import GroupedSelectControl from 'components/grouped-select-control';
import CustomSelectControl from 'components/custom-select-control';

...

const [ option, setOption ] = React.useState< any >();
const [ phone, setPhone ] = React.useState( '' );

...

<LoadBar />
<PhoneNumberControl
	value={ phone }
	onChange={ setPhone }
	label="Phone Number Control"
/>
<GroupedSelectControl
	label="Grouped Select Control"
	options={ options }
	groups={ groups }
	value={ option }
	onChange={ setOption }
	searchable
	placeholder="Select an option"
/>

<CustomSelectControl
	placeholder="Select an option"
	label="Custom Select Control"
	options={ options }
/>
```
</details>

#### Screenshot and videos

<table><tr>
<td><img width="277" alt="custom-select-control" src="https://user-images.githubusercontent.com/7670276/230898889-1581a59c-1993-4209-a10d-7464d8930f94.png"></td><td>

https://user-images.githubusercontent.com/7670276/230898888-82866f5a-5ee2-4d5f-87a9-23ca96a7b56c.mov
</td><td>

https://user-images.githubusercontent.com/7670276/230898884-d620a982-5dd0-4d7b-b48d-6147815eaccb.mov
</td></tr><tr><td colspan=3>

https://user-images.githubusercontent.com/7670276/230898878-a5bf6d55-6d47-46e8-9cc2-cca353786aa5.mov
</td></tr></table>

-------------------


**Post merge**
